bundle-workflow/src/manifests/manifest.py

```diff
@@ -19,6 +19,28 @@ def from_path(cls, path):
         with open(path, "r") as f:
             return cls.from_file(f)
 
+    @classmethod
+    def compact(cls, d):
+        clean = {}
+        for k, v in d.items():
+            if isinstance(v, dict):
+                nested = cls.compact(v)
+                if len(nested.keys()) > 0:
```

Comment:
If we are just checking whether the dict is empty or not. Can be check like `if nested:`

Reply:
This will teach me copying code from SO :) Fixed.